projects/pastanaga-angular/src/lib/date-picker/date-picker.component.spec.ts

```diff
@@ -259,47 +291,40 @@ describe('DatePickerComponent', () => {
             // === Verify ===
             expect(event.stopPropagation).toHaveBeenCalled();
             expect(event.preventDefault).toHaveBeenCalled();
+            expect(touched).not.toHaveBeenCalled();
         });
 
-        it('should not trap events when popup is not displayed', () => {
+        it('should trap events when input is readonly', () => {
```

Comment:
It should also prevent the user to manually enter a value in the input when readonly, and apply the proper readonly style, which is currently not the case (I tried it in your branch demo).

I think it's just a matter of propagating the readonly property to `pa-input` in the template.

Reply:
You are right.  I botched the merge conflicts when updating my branch with master